pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -673,9 +689,125 @@ func addUserConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 	return nil
 }
 
+// splitArgKeyVal takes a value and returns a pair (key, value) of the argument, or two empty strings if there was not
+// a parsed key/val.
+func splitArgKeyVal(val string, delim string) (string, string) {
+	if splitSubArg := strings.SplitN(val, delim, 2); len(splitSubArg) == 2 {
+		return splitSubArg[0], splitSubArg[1]
+	}
+	return "", ""
+}
+
+// getArgValue will search the passed in interface (arg) for a key that matches the searchArg. If a match is found, it
+// returns the value of the argument, otherwise it returns an empty string.
+func getArgValue(arg interface{}, searchArg string, delim string) string {
+	logrus.Tracef("getArgValue (searchArg: %s, delim: %s) type of %v is %T", searchArg, delim, arg, arg)
+	switch arg := arg.(type) {
+	case []interface{}:
+		logrus.Tracef("getArgValue (searchArg: %s, delim: %s) encountered interface slice %v", searchArg, delim, arg)
```

Comment:
Can all these be debug level instead of trace?

Reply:
I put these to trace because they are VERY noisy when they run, and generally don't need to be included in debug arguments.

I really wish we had a --debug-component to specify/filter specific component debug logs.